controller_manager:
  ros__parameters:
    use_sim_time: true
    update_rate: 100

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    tidybot_base_vel_controller:
      type: velocity_controllers/JointGroupVelocityController

    tidybot_base_pos_controller:
      type: position_controllers/JointGroupPositionController

    gen3_7dof_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    gen3_lite_2f_controller:
      type: joint_trajectory_controller/JointTrajectoryController
  
tidybot_base_vel_controller:
  ros__parameters:
    joints:
      - joint_x
      - joint_y
      - joint_th
    command_interfaces:
      - velocity
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 50
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false

tidybot_base_pos_controller:
  ros__parameters:
    joints:
      - joint_x
      - joint_y
      - joint_th
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 50
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false

gen3_7dof_controller:
  ros__parameters:
    joints:
      - joint_1
      - joint_2
      - joint_3
      - joint_4
      - joint_5
      - joint_6
      - joint_7
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 50
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false

gen3_lite_2f_controller:
  ros__parameters:
    joints:
      - right_finger_bottom_joint
      - right_finger_tip_joint
      - left_finger_bottom_joint
      - left_finger_tip_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 50
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false